### DISCUSSION TITLE
Help using App

### DISCUSSION THREAD

**Discussion by HyakuzaO** [georgedouzas/sports-betting#61] [Q&A]
Hello everyone, can someone please help me? I'm using Python 3.9 as requested, everything is set up correctly, but whenever I use the file it always returns without betting any value as shown in the image. I'll also leave the file I'm using, so if anyone can help me find the error and the solution, I would greatly appreciate it as I would love to execute this project in my environment.
[new 1.txt](https://github.com/georgedouzas/sports-betting/files/14627031/new.1.txt)
![Captura de ecrã de 2024-03-17 10-17-55](https://github.com/georgedouzas/sports-betting/assets/160853868/9af9e9cc-8267-4f94-b82a-a4b22711df9a)

**Comment by HyakuzaO** [comment]
Please don't take offense at my questions, I'm just seeking help, and probably for someone who understands, it's an easy fix. Thank you in advance if anyone helps me.

I use the file attached named New 1.

**Comment by georgedouzas** [comment]
Hi @HyakuzaO,

Please post the content of the file.

**Reply by HyakuzaO** [reply]
from sportsbet.datasets import SoccerDataLoader
from sportsbet.evaluation import ClassifierBettor
from sklearn.compose import make_column_transformer
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.impute import SimpleImputer
from sklearn.pipeline import make_pipeline
from sklearn.preprocessing import OneHotEncoder
from sklearn.multioutput import MultiOutputClassifier
from sklearn.model_selection import TimeSeriesSplit

# Parâmetros
leagues = ['England', 'Scotland', 'Germany', 'Italy', 'Spain', 'France', 'Netherlands', 'Belgium', 'Portugal', 'Turkey', 'Greece']
years = [2017, 2018, 2019, 2020, 2021, 2022, 2023]
odds_type = 'market_maximum'
drop_na_thres = 0.8

# Criação do dataloader
dataloader = SoccerDataLoader({'league': leagues, 'year': years})
X_train, Y_train, O_train = dataloader.extract_train_data(drop_na_thres=drop_na_thres, odds_type=odds_type)
X_fix, _, O_fix = dataloader.extract_fixtures_data()

# Configuração do classificador
classifier = make_pipeline(
    make_column_transformer(
        (OneHotEncoder(handle_unknown='ignore'), ['league', 'home_team', 'away_team']),
        remainder='passthrough'
    ),
    SimpleImputer(),
    MultiOutputClassifier(GradientBoostingClassifier(random_state=5)),
)

# Backtesting
tscv = TimeSeriesSplit(5)
init_cash = 1000
bettor = ClassifierBettor(classifier)
bettor.backtest(X_train, Y_train, O_train, tscv=tscv, init_cash=init_cash)
print(bettor.backtest_results_[['Sharpe Ratio', 'Total Return [%]', 'Testing Period']].mean())

# Previsão e exibição de apostas de valor
value_bets = bettor.bet(X_fix, O_fix)

# Imprime as apostas de valor identificadas
print("Apostas de valor identificadas:")
print(value_bets)

**Comment by HyakuzaO** [comment]
I use that script @georgedouzas

**Reply by georgedouzas** [reply]
Solution: Check that your `X_fix` is empty. Therefore no value bets are predicted. Add `'data_source': ['footballdata']` to your parameters grid: `dataloader = SoccerDataLoader({'league': leagues, 'year': years, 'data_source': ['footballdata']})`. Retry to download the data.

Explanation: The reason is that [FiveThirtyEight](https://projects.fivethirtyeight.com/soccer-predictions/) does not update anymore their data. Therefore no fixtures data can be downloaded: your data source should include only the FootballData data source.
